test/e2e/node/pod_hostnamefqdn.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/* This test check that SecurityContext parameters specified at the
+ * pod or the container level work as intended. These tests cannot be
+ * run when the 'SecurityContextDeny' admission controller is not used
+ * so they are skipped by default.
+ */
+
+package node
+
+import (
+	"crypto/rand"
+	"fmt"
+	"math/big"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+
+	"github.com/onsi/ginkgo"
+)
+
+func generatePodName(base string) string {
+	id, err := rand.Int(rand.Reader, big.NewInt(214748))
```

Comment:
How is 214748 determined ?

Reply:
just some random numbers, I copied this function from somewhere and made the number smaller because my FQDN has to fit in 63 chars. I could not find any utility function to generate random pod name of fix size.